docs/source/schemas/dynamic_columns.rst

```diff
@@ -0,0 +1,263 @@
+memory!
+
+Alternatively, one can use ``getRowsColumnRange``. This takes a collection of rows and key ranges, and returns a map of
+row-keys to BatchingVisitables of column values. These may in turn be traversed using the BatchingVisitable
+API. For example, the code below is an implementation of query 1 for an arbitrary String ``person``:
+
+.. code:: java
+
+    TodoTable todoTable = TodoSchemaTableFactory.of().getTodoTable(tx);
+    Map<TodoTable.TodoRow, BatchingVisitable<TodoTable.TodoColumnValue>> results =
+            todoTable.getRowsColumnRange(
+                    ImmutableList.of(TodoTable.TodoRow.of(person)),
+                    BatchColumnRangeSelection.create(PtBytes.EMPTY_BYTE_ARRAY, PtBytes.EMPTY_BYTE_ARRAY, 1));
+
+    AtomicReference<Todo> element = new AtomicReference<>();
+    BatchingVisitable<TodoTable.TodoColumnValue> visitable = Iterables.getOnlyElement(results.values());
+    visitable.batchAccept(1, item -> {
+        element.set(ImmutableTodo.of(item.get(0).getValue()));
+        return false; // Do not want any more, since we know our first batch contains the one.
+    });
+    return element.get();
+
+
+Data Representation (Cassandra)
+-------------------------------
+
+A row with dynamic columns is stored as a single Cassandra-level row. Recall the Atlas schema:
+
+.. code:: java
+
+    CREATE TABLE atlasete.default__todo (
+        key blob,
+        column1 blob,
+        column2 bigint,
+        value blob,
+        PRIMARY KEY (key, column1, column2)
+    )
+
+``key`` consists of the (encoded) row key; ``column1`` refers to the (encoded) column components, and ``column2``
+refers to the Atlas timestamp at which a write occurred. The clustering order of the table follows the
+``ValueByteOrder`` of the relevant column components.
+
+Conversely, observe that in a conventional (fixed columns) AtlasDB table, ``column1`` identifies which column is being
+encoded in the ``value``.
+
+When Not To Use Dynamic Columns
+-------------------------------
+
+Using (Atlas) dynamic columns creates wide rows in Cassandra, because every dynamic column key and value are stored
+with the same row key. This may add limits to the scalability of the data, because all data for a single row key will
+be stored on a single machine in the cluster.
+
+Generally, we recommend that row sizes are kept in the tens of MBs at most, and also below one million dynamic column
+keys.
+
+Appendix: Sample Query Implementations
+--------------------------------------
+
+We present possible implementations of some of the queries (or generalisations of them) described above.
+Note that an implementation of query 1 has been included in the "Retrieval" section.
+
+Query 3 (Size Lower to Upper, Limit N)
+======================================
+
+Assume the existence of variables lower and upper (inclusive bounds for the size range; 10 and 15 in the sample query),
```

Comment:
variable names should also be in double-backticks (`lower` and `upper` here, and `lowerSize` etc. for Query 6).